docs/source/developer_guide/plugins/plugin_tests.rst

```diff
@@ -1,165 +1,146 @@
 .. _plugin.testing:
 
-Writing tests for a plugin
-==========================
-
-When developing a plugin it is important to write tests. The main concern of running
-tests is that the test environment has to be separated from the production environment
-and care should be taken to avoid any unwanted change to the user's database.
-You may have noticed that ``aiida-core`` has its own test framework for developments.
-While it is possible to use the same framework for the plugins,
-it is not ideal as any tests of plugins has to be run with
-the ``verdi devel tests`` command-line interface.
-Special profiles also have to be set mannually by the user and in automated test environments.
-
-AiiDA ships with tools to simplify tests for plugins.
-The recommended way is to use the `pytest`_ framework, while the `unittest`_ package is also supported.
-Internally, test environments are created and managed by the :py:func:`aiida.manage.fixtures.fixture_manager` defined in :py:mod:`aiida.manage.fixtures`.
+Testing AiiDA plugins
+=====================
 
-.. _pytest: https://pytest.org
-.. _unittest: https://docs.python.org/library/unittest.html
-.. _fixture: https://docs.pytest.org/en/latest/fixture.html
+When developing a plugin it is important to write tests.
+We recommend using the `pytest`_ framework, while he `unittest`_ framework is also supported.
 
-Using the pytest framework
---------------------------
+One concern when running tests for AiiDA plugins is to separate the test environment from your production environment.
+Ideally, each individual test should be run against a fresh AiiDA database.
 
-In this section we will introduce using the ``pytest`` framework to write tests for
-plugins.
+AiiDA ships with tools that take care of this for you. They will:
 
-Preparing the fixtures
-^^^^^^^^^^^^^^^^^^^^^^
+ * start a temporary postgres server
+ * create a new database
+ * create a temporary ``.aiida`` folder
+ * create a test profile
+ * (optional) reset the AiiDA database before every individual test
 
-One important concept of the pytest framework is the `fixture`_.
-A fixture is something that a test requires. It could be a predefined object that the
-test act on, resources for the tests, or just some code you want to run before the
-test starts. Please see pytest's `documentation <https://docs.pytest.org/en/latest/>`_ for details, especially if you are new to writing testes.
+thus letting you focus on testing the functionality of your plugin without having to worry about this separation.
 
+.. note::
+   The overhead for setting up the temporary environment is of the order of a few seconds and occurs only once per test session.
 
-To utilize the ``fixture_manager``, we first need to define the actual fixtures:
+If you prefer to run tests on an existing profile, say ``test_profile``, simply set the following environment variable before running your tests::
 
-.. literalinclude:: conftest.py
+  export AIIDA_TEST_PROFILE=test_profile
```

Comment:
is `AIIDA_TEST_PROFILE` correct? I think I saw you use `TEST_AIIDA_PROFILE` in our own tests

Reply:
Thanks for spotting this - this actually highlights a bit of an issue:

We've been using `TEST_AIIDA_BACKEND` so far in the travis tests, so `TEST_AIIDA_PROFILE` would follow this pattern.
However, the other environment variables, such as `AIIDA_PATH` start with `AIIDA` and if we would like to keep this pattern, now would be a good point to decide.

Perhaps `AIIDA_TEST_PROFILE` and `AIIDA_TEST_BACKEND` would be better?

Reply:
To summarize: we discussed to switch all variables to `AIIDA_...`, and to introduce a warning when running `verdi devel tests` while the `AIIDA_TEST_PROFILE` is set, so that as long as the two testing systems coexist, people don't get confused by how to control each.